### PR TITLE
Issue/4764 self hosted block post error

### LINKED ISSUE #4764: Improve error when we detect the host is blocking POST requests on a self hosted site
# Expected behavior

Show a better error message in the full screen dialog, something like:

```
You host is blocking POST requests, the app needs that to communicate to your site.
Contact your host, blablabla.
```

Also, we must add a button the the FAQ page.

# Actual behavior

Full screen dialog with a generic message.

screenshot-2016-11-09_11 43 30 764

# Steps to reproduce the behavior

Try to sign in our test instance here: http://do.wpmt.co/405

### PR DESCRIPTION
**Pull request by mzorz**
Fixes #4764 

Provides a better description for each error and allows to access the Helpshift help pages from a button in SignInDialogFragment.

To test:
1. sign in to a self hosted site that blocks POST requests (so XML-RPC sign in won't work)
2. get the error screen and check the more specific error.
3. Also, now you can tap on "Tell me more" and it will lead you to a more specific description of the problem in our Helpshift FAQ page or section

Check video here https://cloudup.com/cK3DbBPIyRa

cc @maxme @tonyr59h @aforcier 

Note: please check `build.gradle` file as it points to the branch @aforcier was working on

### CODE REVIEW

**Comment by aforcier** [comment]
The PR corresponding to the FluxC branch being used is https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/224 - we can drop the `build.gradle` change once that's merged.

**Comment by maxme** [comment]
@mzorz can you update the build.gradle file (to use flux `develop` branch)?

**Comment by mzorz** [comment]
ready @maxme 🙇

File: WordPress/src/main/java/org/wordpress/android/ui/accounts/SignInFragment.java [hunk 1]
```diff
@@ -1244,10 +1258,16 @@ public void certificateTrusted() {
                 showGenericErrorDialog(getResources().getString(R.string.xmlrpc_missing_method_error));
                 break;
             case XMLRPC_BLOCKED:
-                // TODO: show a better error message
+                //TODO: use this to help the user a bit:  pass the Helpshift page ID or section ID
+                //on the rest of the error cases in this switch
```
Comment (maxme) [inline WordPress/src/main/java/org/wordpress/android/ui/accounts/SignInFragment.java]: Can you do the extra step and add other FAQ id/section to other errors when applicable?

File: WordPress/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -6,6 +6,7 @@
     <!-- account setup -->
     <string name="xmlrpc_error">Couldn\'t connect. Enter the full path to xmlrpc.php on your site and try again.</string>
     <string name="xmlrpc_missing_method_error">Couldn\'t connect. Required XML-RPC methods are missing on the server.</string>
+    <string name="xmlrpc_post_blocked_error">Couldn\'t connect. Your host is blocking POST requests, and the app needs that in order to communicate with your site. \n Please, contact your host to solve this problem. </string>
```
Comment (maxme) [inline WordPress/src/main/res/values/strings.xml]: We should remove the "\n" and the "Please" from this string.

Comment (mzorz) [inline WordPress/src/main/res/values/strings.xml]: addressed in e1120be

File: WordPress/src/main/java/org/wordpress/android/ui/accounts/SignInFragment.java (hunk 1 above)
Comment (mzorz) [inline WordPress/src/main/java/org/wordpress/android/ui/accounts/SignInFragment.java]: Addressed in d96dabf

File: WordPress/src/main/java/org/wordpress/android/ui/accounts/SignInFragment.java [hunk 3]
```diff
@@ -1249,17 +1258,23 @@ public void certificateTrusted() {
...
-                showGenericErrorDialog(getResources().getString(R.string.xmlrpc_missing_method_error));
+                showGenericErrorDialog(getResources().getString(R.string.xmlrpc_missing_method_error),
+                        MISSING_XMLRPC_METHOD_HELPSHIFT_FAQ_ID,
+                        MISSING_XMLRPC_METHOD_HELPSHIFT_FAQ_SECTION);
                 break;
             case XMLRPC_BLOCKED:
-                //TODO: use this to help the user a bit:  pass the Helpshift page ID or section ID
-                //on the rest of the error cases in this switch
+                // use this to help the user a bit:  pass the Helpshift page ID or section ID
+                // on the rest of the error cases in this switch
```
Comment (maxme) [inline WordPress/src/main/java/org/wordpress/android/ui/accounts/SignInFragment.java]: 👍  ;)

**Comment by maxme** [comment]
:shipit:

**Pull request merged**